### ISSUE TITLE
detect.py `--save-crop` bug

### ISSUE THREAD

**Issue by Zengyf-CVer** [ultralytics/yolov5#5544]
### Search before asking

- [X] I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and found no similar bug report.

### YOLOv5 Component

Detection

### Bug

@glenn-jocher 
I ran this command:
```shell
python detect.py --source ./data/images/bus.jpg --save-crop
```
As a result, in the directories of each category in the crops directory, all the pictures have become folders.

![ksnip_20211107-012859](https://user-images.githubusercontent.com/41098760/140618509-c86e63ec-a6f1-40f9-b2e4-eaa7cb0e5b4e.png)
![ksnip_20211107-012918](https://user-images.githubusercontent.com/41098760/140618511-9cc611b0-c41e-4e3a-b64b-be0ec8d3c260.png)

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by glenn-jocher** [comment]
@Zengyf-CVer thanks for the bug report! Yes I'm able to reproduce the same bug. It's likely related to the recent updates to increment_path(), I'll investigate.

**Comment by glenn-jocher** [comment]
@Zengyf-CVer good news 😃! Your original issue may now be fixed ✅ in PR #5545. To receive this update:
- **[Git](https://github.com/ultralytics/yolov5)** – `git pull` from within your `yolov5/` directory or `git clone https://github.com/ultralytics/yolov5` again
- **[PyTorch Hub](https://pytorch.org/hub/ultralytics_yolov5/)** – Force-reload `model = torch.hub.load('ultralytics/yolov5', 'yolov5s', force_reload=True)`
- **[Notebooks](https://github.com/ultralytics/yolov5/blob/master/tutorial.ipynb)** – View updated notebooks  <a href="https://colab.research.google.com/github/ultralytics/yolov5/blob/master/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/ultralytics/yolov5"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **[Docker](https://hub.docker.com/r/ultralytics/yolov5)** – `sudo docker pull ultralytics/yolov5:latest` to update your image <a href="https://hub.docker.com/r/ultralytics/yolov5"><img src="https://img.shields.io/docker/pulls/ultralytics/yolov5?logo=docker" alt="Docker Pulls"></a>

Thank you for spotting this issue and informing us of the problem. Please let us know if this update resolves the issue for you, and feel free to inform us of any other **issues** you discover or **feature requests** that come to mind. Happy trainings with YOLOv5 🚀!
